c/examples/simple_polaris_client.c

```diff
@@ -41,6 +41,7 @@ int main(int argc, const char* argv[]) {
   P1_printf("Opened Polaris context. Authenticating...\n");
 
   if (Polaris_Authenticate(&context, api_key, unique_id) != POLARIS_SUCCESS) {
+    Polaris_Free(&context);
```

Comment:
Structurally I think it'd be cleaner if the guts of main were their own function so we don't have these early returns  everywhere with the polaris free calls repeated before each return. So polaris free just becomes the last line of the main block and the "Run" function can do what's in the body today.

Reply:
Yeah, we could do that. I actually prefer it this way though because it highlights the fact that calling free is important on these error conditions. That becomes even more important when you put this stuff into a loop in the retry example, where you're going to call init multiple times. This is such a simple application I don't think adding another function really makes things any more clear.

Plus, do you call `Polaris_Init()` inside that new function but leave it to the caller to "know" to call `Polaris_Free()` when the function returns? Or do you have to call init outside the  function in main? In the end I don't really think it adds anything to move things to another function.

Reply:
Pasting @anathan's comment from the bottom of the PR to continue the thread here:
```
My thought is it's more clear to have polaris as a "life of the
application" object where Init and Close and encompassing calls to the
business logic inside. But it's just a matter of taste.
```

In general I would agree with that for libraries where that is the design, but for this library that's not really how init is intended to be used for a real application. In a real application, you need to call free and then call init again to reset the context every time something goes wrong and you loop back around (auth failure, connect failed, etc.). Even though the simple example doesn't have any error handling logic so technically it can get away with calling free just once outside of the logic, I still think it's best to have it illustrate the correct way to do things. Then when you look at the connection retry example after seeing the simple one, it looks very similar.